phoenix-core/src/main/java/org/apache/phoenix/replication/SystemCatalogWALEntryFilter.java

```diff
@@ -58,13 +64,63 @@ public Cell filterCell(final WAL.Entry entry, final Cell cell) {
     if (skipCellFilter) {
       return cell;
     }
-    return isTenantRowCell(cell) ? cell : null;
+
+    if (SchemaUtil.isMetaTable(entry.getKey().getTablename().getName())) {
+      return isTenantRowCellSystemCatalog(cell) ? cell : null;
+    } else {
+      return isTenantRowCellSystemChildLink(cell) ? cell : null;
+    }
   }
 
-  private boolean isTenantRowCell(Cell cell) {
+  /**
+   * is the cell for system catalog a tenant owned.
+   * @param cell hbase cell
+   * @return true if the cell is tenant owned
+   */
+  private boolean isTenantRowCellSystemCatalog(final Cell cell) {
     // rows in system.catalog that aren't tenant-owned
     // will have a leading separator byte
     return cell.getRowArray()[cell.getRowOffset()]
-        != QueryConstants.SEPARATOR_BYTE;
+      != QueryConstants.SEPARATOR_BYTE;
+  }
+
+  /**
+   * is the cell for system child link a tenant owned.
+   * @param cell hbase cell
+   * @return true if the cell is tenant owned
+   */
+  private boolean isTenantRowCellSystemChildLink(final Cell cell) {
+    ImmutableBytesWritable key = new ImmutableBytesWritable(
+      cell.getRowArray(), cell.getRowOffset(), cell.getRowLength());
+    //rows in system.catalog that aren't tenant-owned will
+    // have a leading separator byte
+    boolean isTenantRowCell =
```

Comment:
Does the schema for system.child_link table the same as that for system.catalog? and does this assumption of leading separator byte holds true there as well? If so we can `isTenantRowCellSystemCatalog` and reuse the same function here.

Reply:
The schema remains the same but the system catalog table is not supposed to have tenant-owned data with empty tenant id anymore. Thus, I believe we need different methods.